src/Service/Import/CompteUtilisateur/CompteUtilisateurLoader.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Service\Import\CompteUtilisateur;
+
+use App\Factory\EntrepriseFactory;
+use App\Manager\EntrepriseManager;
+use App\Repository\EntrepriseRepository;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+class CompteUtilisateurLoader
+{
+    private const FLUSH_COUNT = 250;
+    public const METADATA_NB_USERS_CREATED = 'nb_users_created';
+    public const METADATA_ERRORS = 'errors';
+    public const METADATA_ENTREPRISE_ACCOUNT_TO_CREATE = 'account_exists';
+
+    private array $metadata = [
+        self::METADATA_NB_USERS_CREATED => 0,
+        self::METADATA_ERRORS => null,
+        self::METADATA_ENTREPRISE_ACCOUNT_TO_CREATE => null,
+    ];
+
+    public function __construct(
+        private CompteUtilisateurMapper $compteUtilisateurMapper,
+        private EntrepriseFactory $entrepriseFactory,
+        private EntrepriseManager $entrepriseManager,
+        private EntrepriseRepository $entrepriseRepository,
+        private ValidatorInterface $validator,
+        private LoggerInterface $logger,
+    ) {
+    }
+
+    public function load(array $data, array $headers, ?OutputInterface $output = null): void
+    {
+        if ($output) {
+            $progressBar = new ProgressBar($output);
+            $progressBar->start(\count($data));
+        }
+
+        $countEntreprise = 0;
+        $lineNumber = 2;
+        foreach ($data as $item) {
+            $dataMapped = $this->compteUtilisateurMapper->map($headers, $item);
+            if (!empty($dataMapped)) {
+                ++$countEntreprise;
+                if ($output) {
+                    $progressBar->advance();
+                }
+
+                $entreprise = $this->entrepriseFactory->createFromArray($dataMapped);
+                $errors = $this->validator->validate($entreprise);
+
+                if (null !== $entreprise && 0 === $errors->count()) {
+                    if (0 === $countEntreprise % self::FLUSH_COUNT) {
+                        $this->logger->info(sprintf('in progress - %s entreprise saved', $countEntreprise));
```

Comment:
il faut quand même faire la partie du else dans ce cas, non ? sinon tu vas sauter une entreprise ?

Reply:
J'ai pas compris :-) car elle est faite.

Reply:
ce que je veux dire, c'est que si tu passes dans ce if, sauf incompréhension de ma part, tu ne fais pas
`$this->metadata[self::METADATA_ENTREPRISE_ACCOUNT_TO_CREATE][] = $entreprise;`
Donc tu sautes l'entreprise de cet index, non ?